dkg/pedersen/mod.go

```diff
@@ -65,41 +68,50 @@ type Actor struct {
 }
 
 // Setup implement dkg.Actor. It initializes the DKG.
-func (a *Actor) Setup(players mino.Players, pubKeys []kyber.Point, threshold int) error {
-
-	if players.Len() != len(pubKeys) {
-		return xerrors.Errorf("there should be as many players as "+
-			"pubKey: %d := %d", players.Len(), len(pubKeys))
-	}
+func (a *Actor) Setup(co crypto.CollectiveAuthority, threshold int) (kyber.Point, error) {
 
 	if a.startRes.Done() {
-		return xerrors.Errorf("startRes is already done, only one setup call is allowed")
+		return nil, xerrors.Errorf("startRes is already done, only one setup call is allowed")
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	sender, receiver, err := a.rpc.Stream(ctx, players)
+	sender, receiver, err := a.rpc.Stream(ctx, co)
 	if err != nil {
-		return xerrors.Errorf("failed to stream: %v", err)
+		return nil, xerrors.Errorf("failed to stream: %v", err)
 	}
 
-	players.AddressIterator().Seek(0)
-	addrs := make([]mino.Address, 0, players.Len())
-	for players.AddressIterator().HasNext() {
-		addrs = append(addrs, players.AddressIterator().GetNext())
+	co.AddressIterator().Seek(0)
```

Comment:
You don't need to seek an iterator that you're not going to use. The function *creates* an iterator.